core/trino-main/src/main/java/io/trino/operator/TableUpdateOperator.java

```diff
@@ -0,0 +1,150 @@
+        {
+            this.operatorId = operatorId;
+            this.planNodeId = requireNonNull(planNodeId, "planNodeId is null");
+            this.metadata = requireNonNull(metadata, "metadata is null");
+            this.session = requireNonNull(session, "session is null");
+            this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+            this.assignments = requireNonNull(assignments, "assignments is null");
+        }
+
+        @Override
+        public Operator createOperator(DriverContext driverContext)
+        {
+            checkState(!closed, "Factory is already closed");
+            OperatorContext context = driverContext.addOperatorContext(operatorId, planNodeId, TableUpdateOperator.class.getSimpleName());
+            return new TableUpdateOperator(context, metadata, session, tableHandle, assignments);
+        }
+
+        @Override
+        public void noMoreOperators()
+        {
+            closed = true;
+        }
+
+        @Override
+        public OperatorFactory duplicate()
+        {
+            return new TableUpdateOperatorFactory(operatorId, planNodeId, metadata, session, tableHandle, assignments);
+        }
+    }
+
+    private final OperatorContext operatorContext;
+    private final Metadata metadata;
+    private final Session session;
+    private final TableHandle tableHandle;
+    private final Map<String, Domain> assignments;
+    private boolean finished;
+
+    public TableUpdateOperator(OperatorContext operatorContext, Metadata metadata, Session session, TableHandle tableHandle, Map<String, Domain> assignments)
+    {
+        this.operatorContext = requireNonNull(operatorContext, "operatorContext is null");
+        this.metadata = requireNonNull(metadata, "metadata is null");
+        this.session = requireNonNull(session, "session is null");
+        this.tableHandle = requireNonNull(tableHandle, "tableHandle is null");
+        this.assignments = requireNonNull(assignments, "assignments is null");
+    }
+
+    @Override
+    public OperatorContext getOperatorContext()
+    {
+        return operatorContext;
+    }
+
+    @Override
+    public void finish()
+    {
+    }
+
+    @Override
+    public boolean isFinished()
+    {
+        return finished;
+    }
+
+    @Override
+    public boolean needsInput()
+    {
+        return false;
+    }
+
+    @Override
+    public void addInput(Page page)
+    {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public Page getOutput()
+    {
+        if (finished) {
+            return null;
+        }
+        finished = true;
+
+        OptionalLong rowsUpdatedCount = metadata.executeUpdate(session, tableHandle, assignments);
+
+        // output page will only be constructed once,
+        // so a new PageBuilder is constructed (instead of using PageBuilder.reset)
+        PageBuilder page = new PageBuilder(1, TYPES);
```

Comment:
Can we extract this to a separate method? `buildUpdatedCountPage(OptionalLong count)` ?

Reply:
refactored